### DISCUSSION TITLE
Make mounting of the hosts docker.sock optional

### DISCUSSION THREAD

**Discussion by Alex-ala** [community/community#25076] [Actions]
<p>Both the Docker Action and a container job start the Docker container with the hosts /var/run/docker.sock mounted. </p>
<p>What is the reason for this behaviour as it blocks easy use of DinD Images on a self-hosted runner?</p>
<p>I would like to have an option to decide not to mount the hosts Docker socket into the jobs container.</p>
<p>In my case I use a Docker in Docker setup to test Java-Applications with Testcontainers and Maven. Testcontainers needs to be able to start Containers and then access them via their exposed port. </p>
<p>When I use a non-DinD image, Testcontainers is also capable of starting the containers, but not accessing their ports, as they are started on the host itself. </p>
<p>Currently my workaround is to have DinD run on an alternate DOCKER_HOST, so that I use /docker.sock instead of the mounted /var/run/docker.sock inside the container. Its not beatuiful and requires a custom image.</p>

**Comment by chrispat** [comment, marked as answer]
<p><a class="mention" href="/alex-ala">@alex-ala</a> Thank you for your feedback and suggestion.  I have added it to our backlog for future enhancements to container jobs and services.</p>
